src/app/services/canvas.service.ts

```diff
@@ -1230,6 +1207,72 @@ export class CanvasService {
         studentId: string,
         overrideTitlePrefix: string
     ): Promise<boolean> {
+        type OverrideDate = Date | null;
+        type OverrideDates = { unlockAt: OverrideDate; lockAt: OverrideDate; dueAt: OverrideDate };
+
+        /**
+         * Merges two override dates into a single choice.
+         * @param min (default: true) if true, return earliest date. if false, return latest date.
+         * @param preserveDate (default: false) if true, returns the other date if one is null
+         * @returns the merged date or null
+         */
+        function mergeOverrideDate(a: OverrideDate, b: OverrideDate, min = true, preserveDate = false): OverrideDate {
```

Comment:
Given that override has been extensively used in Token ATM, would it be better to have a dedicated ES module that handles override-related functionalities instead of having nested functions? It could also be an attempt to decompose `canvas.service.ts` to separate different concerns.